react/components/Popover.tsx

```diff
@@ -1,108 +1,78 @@
-import { path } from 'ramda'
-import React, { Component, ReactNode } from 'react'
-import { withRuntimeContext } from 'vtex.render-runtime'
-
-import telemarketing from '../telemarketing.css'
+import React, { ReactNode, useRef, useState, useCallback } from 'react'
+import classnames from 'classnames'
+import { useRuntime } from 'vtex.render-runtime'
+import useOutsideClick from '../hooks/useOutsideClick'
+import styles from '../telemarketing.css'
 
 interface Props {
   /** Function that will display the header */
-  renderHeader: () => any
+  renderHeader: () => JSX.Element
   arrowClasses: string
   readonly children: ReactNode
 }
 
 /** Component that shows a content when it´s header is clicked */
-export class Popover extends Component<Props> {
-  public boxRef: any = React.createRef()
-
-  public state = {
-    isBoxOpen: false,
-  }
+const Popover = (props: Props) => {
+  const boxRef = useRef<any>(null)
+  const iconRef = useRef<any>(null)
+  const [isBoxOpen, setBoxOpen] = useState(false)
+  const { renderHeader, children, arrowClasses } = props
 
-  private iconRef: any
+  const {
+    hints: { mobile },
+  } = useRuntime()
 
-  public componentWillUnmount() {
-    this.removeListeners()
-  }
+  const handleOutsideClick = () => setBoxOpen(false)
+  useOutsideClick(boxRef, handleOutsideClick, isBoxOpen)
 
-  public render() {
-    const { renderHeader, children } = this.props
-    const mobile = path(['__RUNTIME__', 'hints', 'mobile'], global)
+  const toggleBox = useCallback(() => setBoxOpen(!isBoxOpen), [])
 
-    const boxPositionStyle = mobile
-      ? {}
-      : {
-          right: this.iconRef && this.iconRef.offsetWidth - 43,
-        }
+  const boxPositionStyle = mobile
+    ? {}
+    : {
+        right: iconRef.current && iconRef.current.offsetWidth - 43,
```

Comment:
You can use operator `&&` and return directly this  object ` { right: iconRef.current && iconRef.current.offsetWidth - 43,...`